File: src/main/java/com/wesker926/leetcode/algorithms/p0709/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/12
// @description: ...
package s1

var diff = 'A' - 'a'

func toLowerCase(s string) string {
	result := make([]byte, 0, len(s))
	for _, c := range s {
		if c >= 'A' && c <= 'Z' {
			result = append(result, byte(c-diff))
		} else {
			result = append(result, byte(c))
		}
	}
	return string(result)
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0709/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0709.s1;

/**
 * @author wesker.gh
 * @date 2021/12/12
 * @description ...
 */
public class Solution {

    private static final int DIFF = 'A' - 'a';

    public String toLowerCase(String s) {
        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            sb.append(c >= 'A' && c <= 'Z' ? (char) (c - DIFF) : c);
        }
        return sb.toString();
    }
}
